geopandas/tests/test_vectorized.py

```diff
@@ -0,0 +1,367 @@
+
+    T = from_shapely(triangles)
+
+    result = getattr(T, attr)(q)
+    expected = [getattr(t, attr)(q) for t in triangles]
+
+    assert [a.equals(b) for a, b in zip(result, expected)]
+
+
+@pytest.mark.parametrize('attr', [
+    pytest.mark.xfail('is_closed'),
+    'is_valid',
+    'is_empty',
+    'is_simple',
+    'has_z',
+    'is_ring',
+])
+def test_unary_predicates(attr):
+    triangles = [shapely.geometry.Polygon([(random.random(), random.random())
+                                           for i in range(3)])
+                 for _ in range(10)]
+
+    T = from_shapely(triangles)
+
+    result = getattr(T, attr)()
+    expected = [getattr(t, attr) for t in triangles]
+
+    assert result.tolist() == expected
+
+
+def test_getitem():
+    points = [shapely.geometry.Point(i, i) for i in range(10)]
+    P = from_shapely(points)
+
+    P2 = P[P.x % 2 == 0]
+    assert len(P2) == 5
+    assert all(p.x % 2 == 0 for p in P2)
+
+    P3 = P[[1, 3, 5]]
+    assert len(P3) == 3
+    assert [p.x for p in P3] == [1, 3, 5]
+
+    P4 = P[1::2]
+    assert len(P4) == 5
+    assert [p.x for p in P4] == [1, 3, 5, 7, 9]
+
+
+@pytest.mark.xfail(reason="We don't yet clean up memory well")
+def test_clean_up_on_gc():
+    psutil = pytest.importorskip('psutil')
+    proc = psutil.Process()
+    mem_1 = proc.memory_info().rss
+    x = np.random.random(1000000)
+    y = np.random.random(1000000)
+    nb = x.nbytes
+    mem_2 = proc.memory_info().rss - mem_1
+    assert mem_2 >= x.nbytes + y.nbytes
+
+    points = points_from_xy(x, y)
+    mem_3 = proc.memory_info().rss - mem_1
+    assert mem_3 >= (x.nbytes + y.nbytes) * 2
+
+    del x, y
+    import gc; gc.collect()
+    mem_4 = proc.memory_info().rss - mem_1
+    assert mem_4 < mem_3
+
+    points2 = points[::2]
+    mem_5 = proc.memory_info().rss - mem_1
+    assert mem_5 < mem_4 + nb  # at most a small increase
+
+    del points
+    import gc; gc.collect()
+    mem_6 = proc.memory_info().rss - mem_1
+    assert mem_6 < mem_5 + nb  # still holding onto most of the data
+
+    del points2
+    import gc; gc.collect()
+    time.sleep(0.1)
+    mem_7 = proc.memory_info().rss - mem_1
+    assert mem_7 <= mem_5
+    assert proc.memory_info().rss - mem_1 < nb
+
+
+def test_dir():
+    assert 'contains' in dir(P)
+    assert 'data' in dir(P)
+
+
+def test_chaining():
+    triangles = [shapely.geometry.Polygon([(random.random(), random.random())
+                                           for i in range(3)])
+                 for _ in range(10)]
+
+    T = from_shapely(triangles)
+
+    assert T.contains(T.centroid()).all()
+
+
+@pytest.mark.parametrize('resolution', [16, 25])
+def test_buffer(resolution):
+    expected = [p.buffer(0.1, resolution=resolution, cap_style=CAP_STYLE.round,
+                         join_style=JOIN_STYLE.round)
+                for p in points]
+    result = P.buffer(0.1, resolution=resolution, cap_style=CAP_STYLE.round,
+                        join_style=JOIN_STYLE.round)
```

Comment:
maybe use a non-default value here for the cap_style/join_style keywords here for one case?